src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +10,35 @@ public class RegistrationServiceImpl implements RegistrationService {
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new InvalidDataException("User cannot be null");
+        }
+        try {
+            if (validate(user)) {
+                storageDao.add(user);
+                return user;
+            } else {
+                return null;
+            }
+        } catch (InvalidDataException e) {
+            throw new RuntimeException(e);
+        }
```

Comment:
returning null is a bad practice in most cases
validate could as well be void, it's a guardian method, if smth is wrong - it will throw exception, if not - proceed with saving user